spacy/kb/candidate.py

```diff
@@ -0,0 +1,109 @@
+import abc
+from typing import List, Union, Callable
+
+
+class BaseCandidate(abc.ABC):
+    """A `BaseCandidate` object refers to a textual mention (`alias`) that may or may not be resolved
+    to a specific `entity_id` from a Knowledge Base. This will be used as input for the entity_id linking
+    algorithm which will disambiguate the various candidates to the correct one.
+    Each candidate (alias, entity_id) pair is assigned a certain prior probability.
+
+    DOCS: https://spacy.io/api/kb/#candidate-init
+    """
+
+    def __init__(
+        self, mention: str, entity_id: Union[int, str], entity_vector: List[float]
```

Comment:
The `entity_vector` feels already pretty specific to the in-memory approach, so we can probably leave that field out of the `BaseCandidate` implementation?

Reply:
Hm, or maybe not, I suppose the current EL implementation needs this, right?

Reply:
[Yes](https://github.com/explosion/spaCy/blob/master/spacy/pipeline/entity_linker.py#L279).